bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/HueBridgeMDNSDiscoveryParticipant.java

```diff
@@ -0,0 +1,138 @@
+    private boolean isAutoDiscoveryEnabled = true;
+
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    @Modified
+    protected void modified(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    private void activateOrModifyService(ComponentContext componentContext) {
+        Dictionary<String, @Nullable Object> properties = componentContext.getProperties();
+        String autoDiscoveryPropertyValue = (String) properties
+                .get(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY);
+        if (autoDiscoveryPropertyValue != null && !autoDiscoveryPropertyValue.isBlank()) {
+            isAutoDiscoveryEnabled = Boolean.valueOf(autoDiscoveryPropertyValue);
+        }
+        String removalGracePeriodPropertyValue = (String) properties.get(CONFIG_PROPERTY_REMOVAL_GRACE_PERIOD);
+        if (removalGracePeriodPropertyValue != null && !removalGracePeriodPropertyValue.isBlank()) {
+            try {
+                removalGracePeriod = Long.parseLong(removalGracePeriodPropertyValue);
+            } catch (NumberFormatException e) {
+                logger.warn("Configuration property '{}' has invalid value: {}", CONFIG_PROPERTY_REMOVAL_GRACE_PERIOD,
+                        removalGracePeriodPropertyValue);
+            }
+        }
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return HueBridgeHandler.SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    public String getServiceType() {
+        return SERVICE_TYPE;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        if (isAutoDiscoveryEnabled) {
+            ThingUID uid = getThingUID(service);
+            if (uid != null) {
+                String host = service.getHostAddresses()[0];
+                String id = service.getPropertyString(MDNS_PROPERTY_BRIDGE_ID);
+                String friendlyName = String.format("%s (%s)", service.getName(), host);
+                return DiscoveryResultBuilder.create(uid) //
+                        .withProperties(Map.of( //
+                                HOST, host, //
+                                Thing.PROPERTY_MODEL_ID, service.getPropertyString(MDNS_PROPERTY_MODEL_ID), //
+                                Thing.PROPERTY_SERIAL_NUMBER, id.toLowerCase())) //
+                        .withLabel(friendlyName) //
+                        .withRepresentationProperty(Thing.PROPERTY_SERIAL_NUMBER) //
+                        .withTTL(120L) //
+                        .build();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(ServiceInfo service) {
+        String id = service.getPropertyString(MDNS_PROPERTY_BRIDGE_ID);
+        if (id != null && !id.isBlank()) {
+            return new ThingUID(THING_TYPE_BRIDGE, id.toLowerCase());
```

Comment:
Maybe you should replace possible unexpected characters in id before using it as thing ID ?

Reply:
AFAIK the `bridgeid` is a hexadecimal which cannot contain unexpected characters.